apps/graphql/src/dataloaders/Itinerariesloader.js

```diff
@@ -58,40 +65,107 @@ const fetchItineraries = async (
 const sanitizeItineraries = (response: ApiResponseType): ItinerariesType[] => {
   const itineraries = response.data;
 
-  const newItinerariesStructure = itineraryMock;
+  return itineraries.map(itinerary => {
+    return {
+      // new structure
+      type: getItineraryType(itinerary.routes),
+      startTime: getDate(itinerary.local_departure, itinerary.utc_departure),
+      endTime: getDate(itinerary.local_arrival, itinerary.utc_arrival),
+      destination: getLocation(
+        itinerary.flyTo,
+        itinerary.cityTo,
+        itinerary.countryFrom && itinerary.countryTo.name,
+        itinerary.countryFrom && itinerary.countryTo.code,
+      ),
+      origin: getLocation(
+        itinerary.flyFrom,
+        itinerary.cityFrom,
+        itinerary.countryFrom && itinerary.countryFrom.name,
+        itinerary.countryFrom && itinerary.countryFrom.code,
+      ),
+      sectors:
+        itinerary.route &&
+        itinerary.routes &&
+        getSectors(itinerary.route, itinerary.routes).map(sector => {
```

Comment:
Why not just pass itinerary.route and itinerary.routes and let the getSectors method handle if it is undefined?

Reply:
Done